blob/blob.go

```diff
@@ -138,3 +139,38 @@ func (b *Blob) UnmarshalJSON(data []byte) error {
 	b.namespace = blob.Namespace
 	return nil
 }
+
+// buildBlobsIfExist takes shares and tries building the Blobs from them.
+// It will build blobs either until appShares will be empty or the first incomplete blob will appear, so in this
+// specific case it will return all built blobs + remaining shares.
+func buildBlobsIfExist(appShares []shares.Share) ([]*Blob, []shares.Share, error) {
+	blobs := make([]*Blob, 0, len(appShares))
+	for {
+		if len(appShares) == 0 {
+			return blobs, nil, nil
+		}
+
+		length, err := appShares[0].SequenceLen()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		amount := shares.SparseSharesNeeded(length)
+		if amount > len(appShares) {
+			return blobs, appShares, nil
+		}
+
+		b, err := parseShares(appShares[:amount])
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// only 1 blob will be created bc we passed the exact amount of shares
+		blobs = append(blobs, b[0])
+
+		if amount == len(appShares) {
+			return blobs, nil, nil
+		}
```

Comment:
Looks like this if could be removed, since len(appShares) == 0  will handle same case.

Reply:
otherwise, it could panic here:
`appShares = appShares[amount:]`

Reply:
If amount == 0 it should not panic